src/dynamic_boundary_conditions/hirds_gauges.py

```diff
@@ -0,0 +1,89 @@
+"""
+
+import requests
+from requests.structures import CaseInsensitiveDict
+import pandas as pd
+from shapely.geometry import Point
+import geopandas as gpd
+import sqlalchemy
+from geoalchemy2 import Geometry
+import geopandas
+import numpy as np
+
+
+def get_hirds_gauges_data() -> pd.DataFrame:
+    """Get gauges information from the hirds website using curl command."""
+    url = "https://api.niwa.co.nz/hirds/sites"
+    headers = CaseInsensitiveDict()
+    headers["Connection"] = "keep-alive"
+    headers["sec-ch-ua"] = '"" Not A;Brand";v="99", "Chromium";v="96", "Google Chrome";v="96""'
+    headers["Accept"] = "application/json, text/plain, */*"
+    headers["sec-ch-ua-mobile"] = "?0"
+    headers["User-Agent"] = "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko)\
+        Chrome/96.0.4664.45 Safari/537.36"
+    headers["sec-ch-ua-platform"] = "Windows"
+    headers["Origin"] = "https://hirds.niwa.co.nz"
+    headers["Sec-Fetch-Site"] = "same-site"
+    headers["Sec-Fetch-Mode"] = "cors"
+    headers["Sec-Fetch-Dest"] = "empty"
+    headers["Referer"] = "https://hirds.niwa.co.nz/"
+    headers["Accept-Language"] = "en-GB,en-US;q=0.9,en;q=0.8"
+
+    response = requests.get(url, headers=headers)
+    gauges = pd.read_json(response.text)
+    geometry = [Point(xy) for xy in zip(gauges['longitude'], gauges['latitude'])]
+    crs = 'EPSG:4326'
+    hirds_gauges = gpd.GeoDataFrame(gauges, crs=crs, geometry=geometry)
+    return hirds_gauges
+
+
+def hirds_gauges_to_db(engine, hirds_gauges: geopandas.GeoDataFrame):
+    """Get gauges information from the hirds website and store it in the database."""
+    if sqlalchemy.inspect(engine).has_table("hirds_gauges") is False:
+        print("Storing gauges data from hirds in the database.")
+        hirds_gauges.to_postgis('hirds_gauges', engine, if_exists='replace',
+                                index=False, dtype={'geometry': Geometry(geometry_type='POINT', srid=4326)})
+    else:
+        print("hirds gauges information exists in the database.")
+
+
+def get_new_zealand_boundary(engine) -> geopandas.GeoDataFrame:
+    """Get the boundary information of the new Zealand from region_geometry table."""
+    query1 = "select geometry as geom from region_geometry where regc2021_v1_00_name='New Zealand'"
+    catchment = geopandas.GeoDataFrame.from_postgis(query1, engine)
+    catchment = catchment.to_crs(4326)
+    return catchment
+
+
+def get_guages_location(engine, catchment: geopandas.GeoDataFrame):
+    """
+    Get the gauges location witihin the catchment area from the database and return in the geopandas format.
+
+    engine: to connect to the database.
+    catchment: get the geopandas dataframe of the catchment area.
+    """
+    # Get gauges within the catchment area.
+    catchment_area = catchment.geom[0]
+    query = f'''select * from public.hirds_gauges f
+        where ST_Intersects(f.geometry, ST_GeomFromText('{catchment_area}', 4326))'''
```

Comment:
```suggestion
    query = f'''select * from public.hirds_gauges f
        where ST_Intersects(f.geometry, ST_GeomFromText('{catchment_area}', 4326))'''
```
Was that "f" supposed to be there?

Reply:
yes, I have set an alias for the table. So public.hirds_gauges name is set to 'f', for this particular query. just for simplicity.
As you can see in the next line, I've used f.geometry instead of hirds_gauages.geometry. 
Should I keep hirds_gauges instead of 'f', let me know. :)

Reply:
Ah I see, I wasn't familiar with that syntax. Would this be okay?
```suggestion
    query = f'''select * from public.hirds_gauges hg
        where ST_Intersects(hg.geometry, ST_GeomFromText('{catchment_area}', 4326))'''
```
I am not sure on the conventions here, but the letters then correspond to the table. Or keep it as is if you prefer

Reply:
Yup, makes sense 👍